### PR TITLE
[RA1 Ch03/04] Cloud Deployment Model + Table Formats

### PR DESCRIPTION
**Pull request by pgoyal01**
Fixes Issue #2775

### CODE REVIEW

**Comment by pgoyal01** [comment]
@collivier In Chapter04 it appears that you converted some tables and their rendering shows up but I used the table format it doesn't. Any insights into why? Thanks

**Review by walterkozlowski — APPROVED** [review]

**Comment by collivier** [comment]
@pgoyal01 checking - I second the idea to leverage list-table directive
https://github.com/cntt-n/CNTT/pull/2964/commits/e50934c235fde810d25ac92e55edd7f791af8fe9

File: doc/ref_arch/openstack/chapters/chapter03.rst [hunk 1]
```diff
@@ -696,6 +709,8 @@ characteristics of each. Ultimately the decision rests with the operator
 to achieve specific availability target taking into account use case,
 data centre capabilities, economics and risks.
 
+**Topology Overview**
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter03.rst:712]: @pgoyal01 Is Topology Overview a title? It seems not at the right place, there is another Topology Overview below.

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter03.rst:712]: @karinesevilla That is why I mead this just bolded text

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 2]
```diff
@@ -1086,92 +1080,42 @@ Factors affecting controller node load include number of compute nodes
...
+* - neutron server (neutron-server and neutron-*-plugin)
+  - Manages user requests and exposes the Neutron APIs
+  - Required
+  - Controller node
+* - DHCP agent (neutron-dhcp-agent)
+  - Provides DHCP services to tenant networks and is responsible for maintaining DHCP configuration. For High availability, multiple DHCP agents can be assigned.
+  - Optional depending upon plug-in
+  - Network node (Controller node if no network node present)
+* - L3 agent (neutron-l3-agent)
+  - Provides L3/NAT forwarding for external network access of servers on tenant networks and supports services such as Fire wall-as-a-service (FWaaS) and Load Bala ncer-as-a-service (LBaaS)
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1107]:
```suggestion
  - Provides L3/NAT forwarding for external network access of servers on tenant networks and supports services such as Firewall-as-a-service (FWaaS) and Load Balancer-as-a-service (LBaaS)
```

**Comment by pgoyal01** [comment]
Added a small writeup on edge deployments

File: doc/ref_arch/openstack/chapters/chapter03.rst [hunk 3]
```diff
@@ -728,166 +741,49 @@ Assumptions and conventions:
...
+  - >=2
+  - >=2
+  - >=2
+  - >99.nn
+  - Required
+  - Suitable where local and region application HA is required Control plane could be kept available in one site during 
+    upgrades
+
+
+**Topology Overview**
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter03.rst:784]: @pgoyal01 "Topology overview" can be removed here, as it was added above. And the text following the table is an explanation of the table. For me, there is no need of title here.

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 4]
```diff
@@ -1653,91 +1540,59 @@ subset of the controller services are also deployed on the Intermediate
...
+  - Image Management
+  - Compute
+  - Network Management
+  - Storage Management
+* - CCP
+  - Centralised DC – control nodes
+  - heat-api, heat-engine, nova-placement-api
+  - Identity Provider (IdP), Keystone API
+  - Glance API, Glance Registry
+  - nova-compute api, nova- scheduler, nova-conductor
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1560]: @pgoyal01 There are some extra spaces: nova- scheduler

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 5]
```diff
@@ -1653,91 +1540,59 @@ subset of the controller services are also deployed on the Intermediate
...
+  - neutron -server, neutron-dhcp-agent, neutron-L2-agent, neutron-L3 -agent (optional), neutron-metadata -agent
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1561]: extra spaces: neutron -server, neutron-L3 -agent (optional), neutron-metadata -agent

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 6]
```diff
@@ -1653,91 +1540,59 @@ subset of the controller services are also deployed on the Intermediate
...
+  - Cinder API, Cinder Scheduler, Cinder Volume
+* - DCP: combination of services depending upon Center size
+* - Any DC - Control nodes Option 1
+* - heat-api, heat-engine, nova-placement-api
+* - Identity Provider (IdP), Keystone API
+* - Glance API, Glance Registry
+* - nova-compute api, nova- scheduler, nova-conductor
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1568]: extra space: nova- scheduler

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 7]
```diff
@@ -1653,91 +1540,59 @@ subset of the controller services are also deployed on the Intermediate
...
+* - neutron -server, neutron-dhcp-agent, neutron-L2-agent, neutron-L3-agent (optional), neutron-metadata -agent
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1569]: extra spaces: neutron -server, neutron-metadata -agent

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 8]
```diff
@@ -1653,91 +1540,59 @@ subset of the controller services are also deployed on the Intermediate
...
+  - in one of the DC
+  - in one of ther DC
+  - in one of the DC
+* - CCP or DCP
+  - Compute nodes
+  -  
+  -  
+  -  
+  - nova-compute -agent
+  - neutron-L2- agent, neutron-L3-agent (optional)
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1585]: extra space: neutron-L2- agent

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 9]
```diff
@@ -1653,91 +1540,59 @@ subset of the controller services are also deployed on the Intermediate
...
+  - in the Large DC
+  - in one of the DC
+  - in one of ther DC
+  - in one of the DC
+* - CCP or DCP
+  - Compute nodes
+  -  
+  -  
+  -  
+  - nova-compute -agent
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1584]: extra space: nova-compute -agent

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 10]
```diff
@@ -1653,91 +1540,59 @@ subset of the controller services are also deployed on the Intermediate
...
+  - neutron-L2- agent, neutron-L3-agent (optional)
+  -  
+* - CCP
+  - Compute nodes
+  - nova-placement-api
+  -  
+  -  
+  - nova-compute-agent, nova-conductor
+  - neutron -server, neutron-dhcp-agent, neutron-L2-agent, neutron-L3-agent (optional)
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1593]: extra space: neutron -server

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 11]
```diff
@@ -1773,8 +1628,24 @@ may need to be made for certain Edge clouds depending upon their
...
 affect the definition of OpenStack flavors.
 
-Edge Cloud Deployment
-^^^^^^^^^^^^^^^^^^^^^
+The previous section listed the OpenStack services deployed on the controller nodes depepnding upon
+the control plane distribution. As specified earlier in this chapter, at least 3 controller nodes should be deployed 
+for HA.  Compute nodes may also exist at the sites where controller nodes are deployed.  
+
+Control plane services are not hosted at edge sites. Each edge site can be treated as its own OpenStack AZ. The 
+compute nodes, will host `nova-compute`, a component of the the CCompute Service (Nova), and `neutron-L2-agent`,
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1636]: typo: CCompute

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 12]
```diff
@@ -1773,8 +1628,24 @@ may need to be made for certain Edge clouds depending upon their
 resource capabilities. It should be noted that none of these changes
 affect the definition of OpenStack flavors.
 
-Edge Cloud Deployment
-^^^^^^^^^^^^^^^^^^^^^
+The previous section listed the OpenStack services deployed on the controller nodes depepnding upon
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1631]: Typo: depepnding

**Comment by pgoyal01** [comment]
> @pgoyal01 can I update your PR? they are a couple of blank chars which forbid gating your PR.

@collivier Yes please and can you point to an example. Thanks

**Comment by collivier** [comment]
@pgoyal01 yes that's a proposal. the diff will be the best way to highlight indent and blank char.

File: doc/ref_arch/openstack/chapters/chapter04.rst (hunk 11 above)
Comment (ildikov) [reply doc/ref_arch/openstack/chapters/chapter04.rst:1636]: I don't think there is a 'neutron-L2-agent'. There is an ML2 Plugin on the controller node, but they are using a 'Linux Bridge Agent' on the Computes, see here: https://docs.openstack.org/neutron/train/install/overview.html

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 13]
```diff
@@ -1773,8 +1628,24 @@ may need to be made for certain Edge clouds depending upon their
...
+the control plane distribution. As specified earlier in this chapter, at least 3 controller nodes should be deployed 
+for HA.  Compute nodes may also exist at the sites where controller nodes are deployed.  
+
+Control plane services are not hosted at edge sites. Each edge site can be treated as its own OpenStack AZ. The
```
Comment (ildikov) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1635]: Just to double check. Is that clarified in RM or in this doc that the control plane services are deployed centrally to follow telecom requirements as opposed to being an OpenStack limitation?

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 14]
```diff
@@ -1773,8 +1628,24 @@ may need to be made for certain Edge clouds depending upon their
...
+Control plane services are not hosted at edge sites. Each edge site can be treated as its own OpenStack AZ. The 
+compute nodes, will host `nova-compute`, a component of the the CCompute Service (Nova), and `neutron-L2-agent`,
+a component of the Network Service (Neutron).
+
+The Edge sites may or may not contain local storage. If the edge sites contain storage, then the Block Storage 
+service (Cinder) is usually deployed to run in an active/active mode with the centrally deployed Block Storage 
+service. Instance images are downloaded and stored locally; they can be downloaded even prior to use.
+
+If the edge site doesn't contain storage then the images would need to be cached from the central site. The instance
+images would be downloaded and cached in the Nova cache on first use; they will then be available for subsequent
```
Comment (ildikov) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1644]: Nova also supports to pre-cache images, but they introduced that in the Ussuri release. Not sure if we want to reflect that here? https://docs.openstack.org/nova/latest/admin/image-caching.html#image-pre-caching

**Review by ildikov — COMMENTED** [review]
Just a few minor comments in line.

Thanks,
Ildikó

File: doc/ref_arch/openstack/chapters/chapter04.rst (hunk 13 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter04.rst:1635]: @ildikov RM doesn't deal with deployment and so this control plane distribution is only in the RA.

File: doc/ref_arch/openstack/chapters/chapter04.rst [hunk 15]
```diff
@@ -1773,8 +1645,37 @@ may need to be made for certain Edge clouds depending upon their
 resource capabilities. It should be noted that none of these changes
 affect the definition of OpenStack flavors.
 
-Edge Cloud Deployment
-^^^^^^^^^^^^^^^^^^^^^
+The previous section listed the OpenStack services deployed on the controller
+nodes depending upon the control plane distribution. As specified earlier in
+this chapter, at least 3 controller nodes should be deployed  for HA.  Compute
```
Comment (ildikov) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1650]: There are some extra white spaces in this line. Probably good to fix if there's anything else that needs a new commit in this PR.

**Review by ildikov — APPROVED** [review]
Looks good overall, thanks for the quick updates!

My comment in line is not blocking.

Thanks,
Ildikó

File: doc/ref_arch/openstack/chapters/chapter04.rst (hunk 15 above)
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter04.rst:1650]:
```suggestion
this chapter, at least 3 controller nodes should be deployed for HA.  Compute
```

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter04.rst]: @ildikov Thanks - Fixed.

**Review by karinesevilla — APPROVED** [review]

**Review by walterkozlowski — APPROVED** [review]

**Pull request merged**